File: ScriptService/Dto/Patches/Patch.cs
namespace ScriptService.Dto.Patches {
    
    /// <summary>
    /// provides helper methods to create patches
    /// </summary>
    public static class Patch {

        /// <summary>
        /// creates a replace patch (set a value)
        /// </summary>
        /// <param name="property">property to set</param>
        /// <param name="value">value to set</param>
        /// <returns>patch operation to send to patch endpoints</returns>
        public static PatchOperation Replace(string property, object value) {
            return new PatchOperation {
                Op = "replace",
                Path = $"/{property.ToLower()}",
                Value = value
            };
        }
        
        /// <summary>
        /// adds items to a collection
        /// </summary>
        /// <param name="property">path to collection</param>
        /// <param name="value">item or collection to add</param>
        /// <returns>patch operation to send to patch endpoints</returns>
        public static PatchOperation Add(string property, object value) {
            return new PatchOperation {
                Op = "add",
                Path = $"/{property.ToLower()}",
                Value = value
            };
        }

        /// <summary>
        /// removes items from a collection
        /// </summary>
        /// <param name="property">path to collection</param>
        /// <param name="value">item or collection to add</param>
        /// <returns>patch operation to send to patch endpoints</returns>
        public static PatchOperation Remove(string property, object value) {
            return new PatchOperation {
                Op = "remove",
                Path = $"/{property.ToLower()}",
                Value = value
            };
        }
    }
}

File: nc-scripts/src/app/helpers/patch.ts
import { PatchOperation } from '../dto/patchoperation';

/**
 * provides methods to use when interacting with {@link PatchOperation}s
 */
export class Patch {

    /**
     * creates a new replace patch
     * @param property name of property to patch
     * @param value new property value to apply
     */
    public static replace(property: string, value: any):PatchOperation {
        return {
            op: "replace",
            path: `/${property.toLowerCase()}`,
            value: value
        };
    }

    /**
     * creates patch operations which reflect the difference between two objects
     * current this only creates 'replace' operations for every property which is different
     * from the original
     * @param original object which contains original property values
     * @param changed object of same type as original which contains changed property values
     */
    public static generatePatches(original: any, changed: any): PatchOperation[] {
        let operations: PatchOperation[]=[];

        for(const [key,value] of Object.entries(original)) {
            if(value!==changed[key])
                operations.push(this.replace(key, changed[key]));
        }

        return operations;
    }
}